features/apimgt/org.wso2.carbon.apimgt.publisher.feature/src/main/resources/publisher/source/src/app/data/APIClient.js

```diff
@@ -16,33 +16,35 @@
  * under the License.
  */
 "use strict";
-import Swagger from 'swagger-client'
+import SwaggerClient from 'swagger-client'
 import AuthManager from './AuthManager'
+import Utils from "./Utils";
 
 /**
  * This class expose single swaggerClient instance created using the given swagger URL (Publisher, Store, ect ..)
  * it's highly unlikely to change the REST API Swagger definition (swagger.json) file on the fly,
  * Hence this singleton class help to preserve consecutive swagger client object creations saving redundant IO operations.
  */
-class SingleClient {
-    /**
-     * Check for already created instance of the class, in the `SingleClient._instance` variable,
-     * and return single instance if already exist, else assign `SingleClient._instance` to current instance and return
-     * @param {{}} args : Accept as an optional argument for SwaggerClient constructor.Merge the given args with default args.
-     * @returns {SingleClient|*|null}
+class APIClient {
+    /**--------
+     * Check for already created instance of the class, in the `APIClient._instance` variable,
+     * and return single instance if already exist, else assign `APIClient._instance` to current instance and return
+     * @param {{}} args : Accept as an optional argument for APIClient constructor.Merge the given args with default args.
+     * @returns {APIClient|*|null}
      */
-    constructor(args = {}) {
-        if (SingleClient._instance) {
-            return SingleClient._instance;
-        }
+    constructor(host, args = {}) {
+        host = host || location.host;
+        this.host = host;
```

Comment:
Can't we merge line #36 and #37 ? 

```javascript
this.host = host || location.host;
```

Reply:
Fix with the commit https://github.com/RenukaFernando/carbon-apimgt/commit/685f9e552b3bb256db42d4840496ce259ccd3237